tools/autograd/templates/Functions.cpp

```diff
@@ -408,6 +408,27 @@ Tensor smooth_l1_loss_double_backward_grad_output(const Tensor & grad, const Ten
   return (r * grad).sum().view({1});
 }
 
+Tensor diag_backward(const Tensor & grad, const Tensor & self, int64_t diagonal) {
```

Comment:
We shouldn't keep `self` alive - its sizes would be sufficient and I'm pretty sure there is a way to make the codegen save only them

Reply:
specifying `self.sizes()` in the backward should work.

Reply:
Thanks! `self.sizes()` worked